parsec/core/mountpoint/winfsp_runner.py

```diff
@@ -20,80 +22,56 @@
 logger = get_logger()
 
 
-async def cleanup_broken_links(path: trio.Path) -> None:
-    def target():
-        for child in sync_path.iterdir():
-            try:
-                if not child.exists():
-                    child.unlink()
-            except OSError:
-                pass
-
-    # This should be migrated to trio.Path API once issue #1308 is fixed.
-    sync_path = Path(str(path))
-    await trio.to_thread.run_sync(target)
-
-
-async def is_path_available(path: trio.Path) -> bool:
-    # The path already exists
-    if await path.exists():
-        return False
-
-    # The path is a broken link
-    try:
-        await path.lstat()
-        return True
-    except OSError:
-        pass
-
-    # The path is available
-    return True
-
-
-async def _bootstrap_mountpoint(base_mountpoint_path: Path, workspace_name) -> Path:
-    # Mountpoint can be a drive letter, in such case nothing to do
-    if str(base_mountpoint_path) == base_mountpoint_path.drive:
-        return
-
-    # On Windows, only mountpoint's parent must exists
-    trio_base_mountpoint_path = trio.Path(base_mountpoint_path)
-    await trio_base_mountpoint_path.mkdir(exist_ok=True, parents=True)
+BASE = "PQRSTUVWXYZHIJKLMNO"
+assert len(BASE) == 19
 
-    # Clean up broken links in base directory
-    await cleanup_broken_links(trio_base_mountpoint_path)
 
-    # Find a suitable path where to mount the workspace. The check we are doing
-    # here are not atomic (and the mount operation is not itself atomic anyway),
-    # hence there is still edgecases where the mount can crash due to concurrent
-    # changes on the mountpoint path
-    for tentative in count(1):
-        if tentative == 1:
-            dirname = workspace_name
-        else:
-            dirname = f"{workspace_name} ({tentative})"
+def candidates(index, length, nearest=5):
+    assert 0 <= index < length
+    length = ((length - 1) // nearest + 1) * nearest
+    assert math.gcd(length, len(BASE)) == 1
+    result = ""
+    for _ in BASE:
+        result += BASE[index % len(BASE)]
+        index += length
+    return result
 
-        mountpoint_path = base_mountpoint_path / dirname
-        trio_mountpoint_path = trio_base_mountpoint_path / dirname
 
+async def _get_available_drive(index, length) -> Path:
+    drives = [Path(f"{letter}:\\") for letter in candidates(index, length)]
```

Comment:
What if there is no valid letter ? (I guess we should return instead an `Optional[Path]`, and have caller code raise an exception if None is returned)

Reply:
Good point, we probably want a specific exception for that.